src/Service/Main.php

```diff
@@ -424,20 +489,62 @@ private function populateAssets(Recipe $recipe) {
             mkdir($browserConfigAssetsPath, 0755, true);
         }
         file_put_contents($browserConfigAssetsPath.'/config.php', $browserConfigContents);
+    }
 
-        if ($recipe->includeXdebug || $recipe->developer) {
-            try {
-                $xdebugContents = $this->twig->render('@docker/install-xdebug.sh.twig', ['mode' => $recipe->xdebugMode ?? 'debug']);
-            } catch (\Exception $e) {
-                throw new Exception('Failed to parse install-xdebug.sh template: '.$e->getMessage());
+    private function processConfigFileFromConfig(Recipe $recipe) {
```

Comment:
This isn't quite right as per the ticket.
The ticket states "For both options a / b the config.php file that ends up in the docker container will need to include the config file at the end just before setuplib inclusion. See templates/moodle/config.php.twig"

What that means is that if your recipe specifies $recipe->configFile then the twig config template should just include it. There's no need to process the config file specified in $recipe->configFile.

Twig config template should be doing something like this:

```

{% if (recipeConfig) %}

require_once({{recipeConfig}});

{% endif %}

require_once(__DIR__ . '/lib/setup.php');

// There is no php closing tag in this file,
// it is intentional because it prevents trailing whitespace problems!